python/examples/core/transform.py

```diff
@@ -10,306 +11,377 @@
 
 
 class _TransformThenDescriptor:
-  """Python descriptor dispatching `then` on the `Transform` class as either
-  class or instance method."""
-
-  # Putting implementations in nested functions here directly instead of methods
-  # of Transformation may not be the most elegant approach, but it seems to be
-  # the only one that is picked up by the documentation and typecheck tooling.
-  def __get__(self, obj: tp.Any, objtype: type = None):
-    """Descriptor getter returns either a class-level composition using a
-    metaclass or an instance-level composition."""
-    # Calling as class method.
-    if obj is None:
-      assert issubclass(objtype, Transform)
-
-      def then_class(
-          other_cls: tp.Union[tp.Type['Transform'],
-                              tp.Type['TransformationList']]
-      ) -> tp.Type['TransformationList']:
-        """Creates a new TransformationList subclass chaining this
-        transformation with those provided as argument.
-
-        The new list can be further composed. An instance of the list can be
-        created by passing all parameters required by individual
-        transformations as keyword arguments to the list constructor.
-        """
-        assert issubclass(other_cls, (Transform, TransformationList))
-        name = objtype.__name__ + "Then" + other_cls.__name__
-        transforms = ([objtype, other_cls] if issubclass(other_cls, Transform)
-                      else [objtype] + other_cls._transform_classes)
-        return TransformListMetaclass(name, (TransformationList,), {},
-                                      transforms=transforms)
+    """Python descriptor dispatching `then` on the `Transform` class as either
```

Comment:
This file has unfortunately too many trivial spacing changes that makes it hard to review more deeply.

Please note the autoformatting setup that we've been using:
https://github.com/nicolasvasilache/nicolas.vasilache.github.io/blob/master/vscode_settings/settings.json#L43

Maybe that will help you reduce the amount of changes so this load-bearing file becomes reviewable?

Reply:
Good to know, I was using Black and that file was reformatted like that. I will try your vscode settings. Does it apply the formatting every time you modify a python file?

Reply:
Also, I never heard of YAPF :) Let me try it out !

Reply:
yes, the config I shared reformats on save

Reply:
I tried YAPF out, but that file still seems to suffer for reformatting. Should I try to add this as an independent change as well?

Reply:
Could you please slice this out in a separate PR, address and rebase ?
This way we can land this bigger refactoring.